ui/apps/dashboard/src/components/Billing/Plans/utils.tsx

```diff
@@ -0,0 +1,95 @@
+import { getPeriodAbbreviation } from '@inngest/components/utils/date';
+
+import type { BillingPlan, Entitlements } from '@/gql/graphql';
+
+export type Plan = Omit<BillingPlan, 'entitlements' | 'features'> & {
+  entitlements: Partial<Entitlements>;
+};
+
+export enum PlanNames {
+  Free = 'Free Tier',
+  Basic = 'Basic',
+  Pro = 'Pro',
+  Enterprise = 'Enterprise',
+}
+
+export function processPlan(plan: Plan) {
+  const { name, amount, billingPeriod, entitlements } = plan;
+
+  const featureDescriptions = getFeatureDescriptions(name, entitlements);
+
+  return {
+    name,
+    price: new Intl.NumberFormat('en-US', {
+      style: 'currency',
+      currency: 'USD',
+      maximumFractionDigits: 0,
+    }).format(amount / 100),
+    billingPeriod: typeof billingPeriod === 'string' ? getPeriodAbbreviation(billingPeriod) : 'mo',
+    features: featureDescriptions,
+  };
+}
+
+function getFeatureDescriptions(planName: string, entitlements: Record<string, any>): string[] {
```

Comment:
We never want to use `any` because it completely disables type safety, causing us to miss bugs. For example, `entitlements.runCount.limit` is actually nullable but the `any` hides that, resulting in "0 runs/mo" for plans without a run count limit:
<img width="1225" alt="image" src="https://github.com/user-attachments/assets/d7c3c095-14cd-4afa-8eed-af3fbd324f05">